### PR TITLE
Variable standarization

### PR DESCRIPTION
**Pull request by rsevilla87**
This PR aims to standardize the way to pass certain variables to the workloads.

### CODE REVIEW

**Comment by bengland2** [comment]
totally agree with this one, smallfile still requires es-index, there is no reason for this because it should just default to ripsaw-smallfile, I thought it was just smallfile that had the problem.    Make life simpler for people that use ripsaw and snafu.

File: roles/fio-distributed/templates/client.yaml [hunk 1]
```diff
@@ -51,7 +47,7 @@ spec:
 {% for numjobs in fiod.numjobs %}
 {% for bs in fiod.bs %}
 {% for job in fiod.jobs %}
-             cat /tmp/fio/fiojob-{{job}}-{{bs}}-{{numjobs}}; mkdir -p /tmp/fiod-{{ uuid }}/fiojob-{{job}}-{{bs}}-{{numjobs}}; python /opt/snafu/run_snafu.py -t fio -H /tmp/host/hosts -j /tmp/fio/fiojob-{{job}}-{{bs}}-{{numjobs}} -s {{fiod.samples}} -d /tmp/fiod-{{ uuid }}/fiojob-{{job}}-{{bs}}-{{numjobs}};
+             cat /tmp/fio/fiojob-{{job}}-{{bs}}-{{numjobs}}; mkdir -p /tmp/fiod-${uuid}/fiojob-{{job}}-{{bs}}-{{numjobs}}; python /opt/snafu/run_snafu.py -t fio -H /tmp/host/hosts -j /tmp/fio/fiojob-{{job}}-{{bs}}-{{numjobs}} -s {{fiod.samples}} -d /tmp/fiod-${uuid}/fiojob-{{job}}-{{bs}}-{{numjobs}};
```
Comment (dry923) [inline roles/fio-distributed/templates/client.yaml:50]: Why the change from {{ uuid }} to ${uuid}? It only seems to be done in fio and uperf

**Comment by rsevilla87** [comment]
@bengland2 I have had to increase the fio test timeout since it was timing out in the CI.

File: resources/crds/ripsaw_v1alpha1_ripsaw_crd.yaml [hunk 2]
```diff
@@ -29,6 +29,8 @@ spec:
                     type: string
                   port: 
                     type: integer
+                  es_index:
```
Comment (dry923) [inline resources/crds/ripsaw_v1alpha1_ripsaw_crd.yaml:32]: Should we rename this to just index as es is redundant since its in the elasticsearch object?

Comment (rsevilla87) [reply resources/crds/ripsaw_v1alpha1_ripsaw_crd.yaml:32]: I named it `es_index` since index is the name of a function in Jinja and will crash.

File: roles/load-ycsb/tasks/main.yml [hunk 3]
```diff
@@ -35,9 +35,9 @@
               - name: es_port
                 value: "{{elasticsearch.port | default()}}"
               - name: es_index
-                value: "{{ es_index }}"
+                value: "{{ elasticsearch.es_index | default('ripsaw-ycsb') }}"
               - name: user
-                value: "{{test_user | default()}}"
+                value: "{{test_user | default('ripsaw') }}"
```
Comment (dry923) [inline roles/load-ycsb/tasks/main.yml:40]: "{{test_user | default('ripsaw')}}" or "{{ test_user | default('ripsaw') }}" for consistency

File: roles/load-ycsb/tasks/main.yml [hunk 4]
```diff
@@ -48,7 +48,7 @@
                 value: "{{uuid}}"
               imagePullPolicy: Always
               command: ["/bin/sh", "-c"]
-              args: ["python /opt/snafu/run_snafu.py --tool ycsb -r 1 -l -d {{ ycsb.driver }} -u {{ uuid }} --user {{ test_user | default(\"ripsaw\") }} -w {{ ycsb_workload }} -x \"{{ ycsb.options_load }}\""]
+              args: ["python /opt/snafu/run_snafu.py --tool ycsb -r 1 -l -d {{ ycsb.driver }} -u {{ uuid }} --user {{ test_user|default('ripsaw') }} -w {{ ycsb_workload }} -x \"{{ ycsb.options_load }}\""]
```
Comment (dry923) [inline roles/load-ycsb/tasks/main.yml:51]: Why are we re-defaulting test_user when we do it in line 40 above?

File: roles/smallfile-bench/templates/workload_job.yml.j2 [hunk 5]
```diff
@@ -40,24 +40,21 @@ spec:
...
-            value: "{{ es_index }}"
-{% endif %}
+            value: "{{ elasticsearch.es_index | default("ripsaw-smallfile") }}"
           - name: redis_host
             value: "{{ bo.resources[0].status.podIP }}"
           - name: clients
             value: "{{ smallfile.clients }}"
           args:
             - python /tmp/smallfile/subscriber {{bo.resources[0].status.podIP}};
-              export TMPDIR=/tmp
```
Comment (dry923) [inline roles/smallfile-bench/templates/workload_job.yml.j2:60]: Are we removing the TMPDIR definition outright? In fio it was re-defined elsewhere but here it just seems to be a remove line

File: roles/uperf-bench/templates/workload.yml.j2 [hunk 6]
```diff
@@ -85,7 +84,7 @@ spec:
 {% for nthr in uperf.nthrs %}
                  cat /tmp/uperf-test/uperf-{{test}}-{{proto}}-{{size}}-{{nthr}};
                  for i in `seq 1 {{uperf.samples}}`; do
-                   python /opt/snafu/run_snafu.py --tool uperf -w /tmp/uperf-test/uperf-{{test}}-{{proto}}-{{size}}-{{nthr}} -r $i --resourcetype {{resource_kind}} -u {{ uuid }} --user {{test_user | default("ripsaw")}};
+                   python /opt/snafu/run_snafu.py --tool uperf -w /tmp/uperf-test/uperf-{{test}}-{{proto}}-{{size}}-{{nthr}} -r $i --resourcetype {{resource_kind}} -u {{uuid}} --user {{test_user|default("ripsaw")}};
```
Comment (dry923) [inline roles/uperf-bench/templates/workload.yml.j2:87]: Same comment re test_user being re-defaulted again

File: tests/test_fiod.sh [hunk 7]
```diff
@@ -27,7 +27,7 @@ function functional_test_fio {
   wait_for "kubectl -n my-ripsaw wait --for=condition=Initialized -l app=fio-benchmark-$uuid pods --timeout=300s" "300s"
   fio_pod=$(get_pod "app=fiod-client-$uuid" 300)
   wait_for "kubectl wait --for=condition=Initialized pods/$fio_pod -n my-ripsaw --timeout=500s" "500s" $fio_pod
-  wait_for "kubectl wait --for=condition=complete -l app=fiod-client-$uuid jobs -n my-ripsaw --timeout=500s" "500s" $fio_pod
+  wait_for "kubectl wait --for=condition=complete -l app=fiod-client-$uuid jobs -n my-ripsaw --timeout=900s" "900s" $fio_pod
```
Comment (dry923) [inline tests/test_fiod.sh:30]: Any idea why this PR would cause a delay here? The 500s timeout seemed to work prior. My concern going to 900s is that it would be done potentially twice in this test and then repeated up to 3 times. Any way to scale this down some?

File: roles/smallfile-bench/templates/workload_job.yml.j2 (hunk 5 above)
Comment (bengland2) [reply roles/smallfile-bench/templates/workload_job.yml.j2:60]: @dry923 , good question.  He probably removed it because TMPDIR isn't referenced anywhere in this file, but subprocesses can reference it too.      in this line in smallfile code, TMPDIR is referenced, but if it's not defined smallfile just uses /var/tmp/, which does always exist, so not a fatal error.   And it doesn't use it for files it accesses during the test, just logging and thread-ready synch. files, so I don't think it matters.

Comment (rsevilla87) [reply roles/smallfile-bench/templates/workload_job.yml.j2:60]: Exactly, that's the reason :). I checked smallfile's source and verified it. TMPDIR is back again as env var. I don't want to break any consistency of the workloads, just reorganize and clean-up.

File: resources/crds/ripsaw_v1alpha1_ripsaw_crd.yaml (hunk 2 above)
Comment (rsevilla87) [reply resources/crds/ripsaw_v1alpha1_ripsaw_crd.yaml]: I changed it to index_name

Comment (rsevilla87) [reply resources/crds/ripsaw_v1alpha1_ripsaw_crd.yaml]: I eventually changed it for `index_name`

**Comment by dry923** [comment]
@rsevilla87 can you revert the change to the fio timeouts now that https://github.com/cloud-bulldozer/ripsaw/pull/334 has been merged?

File: tests/test_crs/valid_fiod.yaml [hunk 8]
```diff
@@ -7,8 +7,9 @@ spec:
   elasticsearch:
     server: "marquez.perf.lab.eng.rdu2.redhat.com"
     port: 9200
+    index_name: ripsaw-fio
   metadata:
-    collection: true
```
Comment (dry923) [inline tests/test_crs/valid_fiod.yaml:11]: collection should be set to true still

File: tests/test_crs/valid_smallfile.yaml [hunk 9]
```diff
@@ -11,7 +11,7 @@ spec:
   elasticsearch:
     server: "marquez.perf.lab.eng.rdu2.redhat.com"
     port: 9200
-  es_index: ripsaw-smallfile
+  index_name: ripsaw-smallfile
```
Comment (dry923) [inline tests/test_crs/valid_smallfile.yaml:14]: needs indentation under elasticsearch

Comment (rsevilla87) [reply tests/test_crs/valid_smallfile.yaml]: my fault, I forgot to revert this change after running tests locally :)

File: tests/test_crs/valid_fiod.yaml (hunk 8 above)
Comment (rsevilla87) [reply tests/test_crs/valid_fiod.yaml:11]: ouch!, I forgot to revert this change after running test locally :)

**Comment by dry923** [comment]
@bengland2 @jtaleric Any concerns with this or are we good to merge?

File: roles/fs-drift/templates/workload_job.yml.j2 [hunk 10]
```diff
@@ -42,29 +42,28 @@ spec:
...
           - name: clustername
             value: "{{ clustername }}"
-{% if elasticsearch.server is defined %}
           - name: es
-            value: "{{ elasticsearch.server }}"
+            value: "{{ elasticsearch.server | default() }}"
           - name: es_port
-            value: "{{ elasticsearch.port }}"
+            value: "{{ elasticsearch.port | default() }}"
           - name: es_index
```
Comment (bengland2) [inline roles/fs-drift/templates/workload_job.yml.j2:52]: shouldn't this default as well?   You know you're in ripsaw at this point so why not default to ripsaw-fs-drift ?

File: roles/fs-drift/templates/workload_job.yml.j2 [hunk 11]
```diff
@@ -42,29 +42,28 @@ spec:
...
             value: "{{ uuid }}"
           - name: test_user
-            value: "{{ test_user }}"
+            value: "{{ test_user | default("ripsaw") }}"
           - name: clustername
             value: "{{ clustername }}"
-{% if elasticsearch.server is defined %}
           - name: es
-            value: "{{ elasticsearch.server }}"
+            value: "{{ elasticsearch.server | default() }}"
```
Comment (bengland2) [inline roles/fs-drift/templates/workload_job.yml.j2:49]: this doesn't make sense, if elasticsearch.server is defined, then why would you need to default it?   I think you meant "if elasticsearch is defined?

File: roles/smallfile-bench/templates/workload_job.yml.j2 [hunk 12]
```diff
@@ -40,24 +40,23 @@ spec:
...
           - name: clustername
             value: "{{ clustername }}"
-{% if elasticsearch.server is defined %}
           - name: es
-            value: "{{ elasticsearch.server }}"
+            value: "{{ elasticsearch.server | default() }}"
           - name: es_port
-            value: "{{ elasticsearch.port }}"
+            value: "{{ elasticsearch.port | default() }}"
           - name: es_index
```
Comment (bengland2) [inline roles/smallfile-bench/templates/workload_job.yml.j2:50]: why do you need to have es_index if value is removed?   Why not just let es_index default to ripsaw-smallfile?  We know we are in ripsaw at this point.   This is one of most annoying parts of smallfile CR at this point, would like to get rid of it.

File: roles/smallfile-bench/templates/workload_job.yml.j2 [hunk 13]
```diff
@@ -40,24 +40,23 @@ spec:
...
             value: "{{ uuid }}"
           - name: test_user
-            value: "{{ test_user }}"
+            value: "{{ test_user | default("ripsaw") }}"
           - name: clustername
             value: "{{ clustername }}"
-{% if elasticsearch.server is defined %}
           - name: es
-            value: "{{ elasticsearch.server }}"
+            value: "{{ elasticsearch.server | default() }}"
```
Comment (bengland2) [inline roles/smallfile-bench/templates/workload_job.yml.j2:47]: same comment as with fs-drift - did you mean "if elasticsearch is defined"?   because it doesn't make sense to have a default for elasticsearch.server if the onlly way you can get to it is if the user defines elasticsearch.server in their CR.

File: roles/fio-distributed/templates/client.yaml [hunk 14]
```diff
@@ -20,21 +20,17 @@ spec:
...
           - name: es
-            value: "{{ es_server }}"
+            value: "{{ elasticsearch.server | default() }}"
           - name: es_port
-            value: "{{ es_port }}"
+            value: "{{ elasticsearch.port | default() }}"
           - name: es_index
-            value: "{{ es_index }}"
-{% endif %}
+            value: "{{ elasticsearch.index_name | default("ripsaw-fio") }}"
```
Comment (bengland2) [inline roles/fio-distributed/templates/client.yaml:33]: can we do this for smallfile and fs-drift also?   Thanks!

File: roles/fs-drift/templates/workload_job.yml.j2 (hunk 10 above)
Comment (rsevilla87) [reply roles/fs-drift/templates/workload_job.yml.j2:52]: I don't understand. It's already set to "ripsaw-fs-drift"

File: roles/smallfile-bench/templates/workload_job.yml.j2 (hunk 12 above)
Comment (rsevilla87) [reply roles/smallfile-bench/templates/workload_job.yml.j2:50]: I don't understand. It's already set to "ripsaw-smallfile"

File: roles/fs-drift/templates/workload_job.yml.j2 (hunk 10 above)
Comment (bengland2) [reply roles/fs-drift/templates/workload_job.yml.j2:52]: this default could be computed from "ripsaw-{{ spec.workload.name }}" once, by ripsaw, not each benchmark.

Comment (rsevilla87) [reply roles/fs-drift/templates/workload_job.yml.j2:52]: We can't do that now since that the workload name doesn't mach with the index name, we would have rename the workloads to do so. I think that should be done in a different PR.

Comment (bengland2) [reply roles/fs-drift/templates/workload_job.yml.j2:53]: didn't know that, ok then I'll withdraw that suggestion.

File: roles/smallfile-bench/templates/workload_job.yml.j2 (hunk 12 above)
Comment (bengland2) [reply roles/smallfile-bench/templates/workload_job.yml.j2:51]: withdrawing this comment, it is really another case of the same comment up above that you answered.

File: roles/fio-distributed/templates/client.yaml (hunk 14 above)
Comment (bengland2) [reply roles/fio-distributed/templates/client.yaml:34]: you answered this question before, am withdrawing this comment.

File: roles/fs-drift/templates/workload_job.yml.j2 (hunk 11 above)
Comment (bengland2) [reply roles/fs-drift/templates/workload_job.yml.j2]: @rsevilla87 what about this one?   This is my last comment - the one below is just another case of it.

Comment (rsevilla87) [reply roles/fs-drift/templates/workload_job.yml.j2]: I've made several updates around this, take a look when you get time.

**Comment by rsevilla87** [comment]
Let's wait for https://github.com/cloud-bulldozer/ripsaw/pull/330 to be merged before this one.

**Comment by dry923** [comment]
@rsevilla87 can you give this guy a rebase please

**Comment by rsevilla87** [comment]
@dry923 done!
